src/routes/auth/providers/utils.ts

```diff
@@ -113,14 +114,28 @@ export const initProvider = <T extends Strategy>(
 
   subRouter.get('/', passport.authenticate(strategyName, { session: false }))
 
-  subRouter.get(
-    '/callback',
-    passport.authenticate(strategyName, {
-      failureRedirect: PROVIDERS_FAILURE_REDIRECT,
-      session: false
-    }),
-    providerCallback
-  )
+  // The Sign in with Apple auth provider requires a POST route for authentication
+  if (strategyName === 'apple') {
```

Comment:
How should we do this? Shall we always expose  POST and GET endpoint or shall we cherry pick similar to the code here?

Reply:
It would be great if we could expose both POST and GET at the same time? Can we chain that somehow in Express?

Reply:
Or we could use `subRouter.all('/callback',[...]`?

I don't think we need to worry about the request method here.